src/compound.rs

```diff
@@ -39,34 +39,36 @@ where
     S: Store,
 {
     type Leaf;
-    type Annotation: Canon<S> + Annotation<Self, S> + Clone + Sized;
+    type Annotation: Canon<S> + Clone + Sized;
 
     fn child(&self, ofs: usize) -> Child<Self, S>;
     fn child_mut(&mut self, ofs: usize) -> ChildMut<Self, S>;
 }
 
-pub trait Nth<'a, S, const N: usize>
+pub trait Nth<'a, S>
 where
     Self: Compound<S> + Sized,
+    Self::Annotation: Annotation<Self, S>,
     S: Store,
 {
-    fn nth(
+    fn nth<const N: usize>(
         &'a self,
         n: u64,
     ) -> Result<Option<Branch<'a, Self, S, N>>, S::Error>;
-    fn nth_mut(
+
+    fn nth_mut<const N: usize>(
```

Comment:
Looks good to me. I also tried this but I didn't want to have both methods different `N`, I wasn't sure if that on API level would be OK. If it's okay then this is will reduce a lot the code.